lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -422,12 +463,92 @@ func (c *ContainerProxy) StreamContainerLogs(name string, out io.Writer, started
 	return nil
 }
 
+// Stop will stop (shutdown) a VIC container.
+//
+// returns
+//	error
+func (c *ContainerProxy) Stop(vc *viccontainer.VicContainer, name string, seconds int, unbound bool) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if c.client == nil {
+		return InternalServerError("ContainerProxy.Stop failed to get a portlayer client")
+	}
+
+	//retrieve client to portlayer
+	handle, err := c.Handle(vc.ContainerID, name)
+	if err != nil {
+		return err
+	}
+
+	// we have a container on the PL side lets check the state before proceeding
+	// ignore the error  since others will be checking below..this is an attempt to short circuit the op
+	// TODO: can be replaced with simple cache check once power events are propagated to persona
+	running, err := c.IsRunning(vc)
+	if err != nil {
+		cache.ContainerCache().DeleteContainer(vc.ContainerID)
```

Comment:
I didn't add that.  I moved that code from container.go to the proxy code and didn't change the logic.  I also questioned that code.  After your questioning, I looked into it and went ahead and change it to only do this if we get a not found from IsRunning.